cmd/root_test.go

```diff
@@ -1536,6 +1536,232 @@ func TestGit(t *testing.T) {
 	)
 }
 
+func TestJujutsu(t *testing.T) {
+	as := require.New(t)
+
+	tempDir := test.TempExamples(t)
+	configPath := filepath.Join(tempDir, "/treefmt.toml")
+
+	test.ChangeWorkDir(t, tempDir)
+
+	// basic config
+	cfg := &config.Config{
+		FormatterConfigs: map[string]*config.Formatter{
+			"echo": {
+				Command:  "echo", // will not generate any underlying changes in the file
+				Includes: []string{"*"},
+			},
+		},
+	}
+
+	test.WriteConfig(t, configPath, cfg)
+
+	// init a jujutsu repo (disable signing for testing)
+	jjCmd := exec.Command("jj", "git", "init", "--config", "signing.backend=none")
+	as.NoError(jjCmd.Run(), "failed to init jujutsu repository")
+
+	// run before adding anything to the index
+	// we shouldn't pick up untracked files, because here the jujutsu walker differs from the git walker
+	treefmt(t,
+		withConfig(configPath, cfg),
+		withNoError(t),
+		withStats(t, map[stats.Type]int{
+			stats.Traversed: 0,
+			stats.Matched:   0,
+			stats.Formatted: 0,
+			stats.Changed:   0,
+		}),
+	)
+
+	// track new files (disable signing for testing), this step is required based on the jujutsu configuration.
+	// I prefer `snapshot.auto-track=none()`, which doesn't track new files automatically.
+	jjCmd = exec.Command("jj", "file", "track", "--config", "signing.backend=none", ".")
+	as.NoError(jjCmd.Run(), "failed to add everything to the index")
+
+	// update jujutsu's index (disable signing for testing)
+	jjCmd = exec.Command("jj", "--config", "signing.backend=none")
+	as.NoError(jjCmd.Run(), "failed to update the index")
+
+	// This is our first pass, since previously the files were not in the index. This should format all files.
+	treefmt(t,
+		withConfig(configPath, cfg),
+		withNoError(t),
+		withStats(t, map[stats.Type]int{
+			stats.Traversed: 32,
+			stats.Matched:   32,
+			stats.Formatted: 32,
+			stats.Changed:   0,
+		}),
+	)
+
+	// create a file which should be in .gitignore
+	f, err := os.CreateTemp(tempDir, "test-*.txt")
+	as.NoError(err, "failed to create temp file")
+
+	// add everything to the index (disable signing for testing), this command shoud ignore files in .gitignore
+	jjCmd = exec.Command("jj", "file", "track", "--config", "signing.backend=none", ".")
+	as.NoError(jjCmd.Run(), "failed to add everything to the index")
+
+	// update jujutsu's index (disable signing for testing)
```

Comment:
> (disable signing for testing)

This is pretty repetitive. Can we instead create a config file for the test?

Reply:
I agree, I'll change that.

Reply:
I created a `test/config` directory containing a basic Jujutsu configuration. This config is included in the test environment by setting the `XDG_CONFIG_HOME` environment variable.